examples/iw_vae_normflow.py

```diff
@@ -225,32 +225,33 @@ def latent_gaussian_x_bernoulli(z, z_mu, psi_u_list, z_log_var, x_mu, x, eq_samp
     """
 
     # reshape the variables so batch_size, eq_samples and iw_samples are separate dimensions
-    z = z.reshape((-1, eq_samples, iw_samples, latent_size))
+    z0 = z0.reshape((-1, eq_samples, iw_samples, latent_size))
+    zk = zk.reshape((-1, eq_samples, iw_samples, latent_size))
     x_mu = x_mu.reshape((-1, eq_samples, iw_samples, num_features))
 
-    # dimshuffle x, z_mu and z_log_var since we need to broadcast them when calculating the pdfs
-    x = x.dimshuffle(0,'x','x',1)                   # size: (batch_size, eq_samples, iw_samples, num_features)
-    z_mu = z_mu.dimshuffle(0,'x','x',1)             # size: (batch_size, eq_samples, iw_samples, num_latent)
-    z_log_var = z_log_var.dimshuffle(0,'x','x',1)   # size: (batch_size, eq_samples, iw_samples, num_latent)
+    for i in range(len(logdet_J_list)):
+        logdet_J_list[i] = logdet_J_list[i].reshape((-1, eq_samples, iw_samples))
 
-    for i in range(len(psi_u_list)):
-        psi_u_list[i] = psi_u_list[i].reshape((-1, eq_samples, iw_samples))
+    # dimshuffle x, z_mu and z_log_var since we need to broadcast them when calculating the pdfs
+    x = x.dimshuffle(0, 'x', 'x', 1)                    # size: (batch_size, eq_samples, iw_samples, num_features)
+    z0_mu = z0_mu.dimshuffle(0, 'x', 'x', 1)            # size: (batch_size, eq_samples, iw_samples, num_latent)
+    z0_log_var = z0_log_var.dimshuffle(0, 'x', 'x', 1)  # size: (batch_size, eq_samples, iw_samples, num_latent)
 
     # calculate LL components, note that the log_xyz() functions return log prob. for indepenedent components separately 
     # so we sum over feature/latent dimensions for multivariate pdfs
-    log_qz_given_x = log_normal2(z, z_mu, z_log_var).sum(axis=3)
-    log_pz = log_stdnormal(z).sum(axis=3)
-    log_px_given_z = log_bernoulli(x, T.clip(x_mu,epsilon,1-epsilon)).sum(axis=3)
+    log_q0z0_given_x = log_normal2(z0, z0_mu, z0_log_var).sum(axis=3)
+    log_pzk = log_stdnormal(zk).sum(axis=3)
```

Comment:
This is the main change `log_stdnormal(z).sum(axis=3)` to `log_stdnormal(zk).sum(axis=3)`.
Most other changes are to clarify code.